src/client/pfs/pfs.proto

```diff
@@ -312,12 +312,20 @@ message TagObjectRequest {
   repeated Tag tags = 2;
 }
 
+message ListObjectsRequest {}
+
+message ListObjectsTaggedWithPrefixRequest {
+  string prefix = 1;
+}
```

Comment:
Also, why not just have the prefix be part of the `ListObjectsReqest`? if it's empty then you just return everything which is what you'd expect from a prefix field.